### PR TITLE
fix Compression to_string and into string impls

### PR DESCRIPTION
**Pull request by conorbros**

### CODE REVIEW

File: cassandra-protocol/src/compression.rs [hunk 1]
```diff
@@ -169,6 +168,28 @@ impl From<String> for Compression {
     }
 }
 
+impl From<Compression> for String {
+    /// It converts `Compression` into `String`. If compression is `None` then empty string will be
```
Comment (krojew) [inline cassandra-protocol/src/compression.rs:172]: It will be NONE, not empty.

File: cassandra-protocol/src/compression.rs [hunk 2]
```diff
@@ -169,6 +168,28 @@ impl From<String> for Compression {
...
+        }
+    }
+}
+
+impl fmt::Display for Compression {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Compression::Lz4 => write!(fmt, "lz4"),
+            Compression::Snappy => write!(fmt, "snappy"),
+            Compression::None => write!(fmt, "none"),
```
Comment (krojew) [inline cassandra-protocol/src/compression.rs:188]: This is inconsistent with the implementation above.

**Review by krojew — CHANGES_REQUESTED** [review]
Why is this needed?

**Comment by conorbros** [comment]
I proposed this change so that the `to_string` function for the `Compression` enum formats it correctly according to the protocol spec. It should be either all caps or lowercase. The default impl will format it as the enum name e.g. `"Snappy"`, when it should be `"snappy"` or `"SNAPPY"`.

**Comment by krojew** [comment]
In Rust typically string representations of enums reflect the names itself. I think it would be better to explicitly add `to_protocol_string()`/`from_protocol_string()` or similar.

File: cassandra-protocol/src/compression.rs [hunk 3]
```diff
@@ -64,7 +63,7 @@ impl Clone for CompressionError {
 }
 
 /// Enum which represents a type of compression. Only non-startup frame's body can be compressed.
-#[derive(Debug, PartialEq, Clone, Copy, Eq, Ord, PartialOrd, Hash, Display)]
+#[derive(Debug, PartialEq, Clone, Copy, Eq, Ord, PartialOrd, Hash)]
```
Comment (krojew) [inline cassandra-protocol/src/compression.rs:66]: You forgot to put back Display.

**Review by krojew — CHANGES_REQUESTED** [review]

**Review by krojew — APPROVED** [review]

**Pull request merged**